i-need-another-coffee/decidim
# Add moderation details page

#### :tophat: What? Why?

This adds two new views to the moderation panel:
- In the moderations list you can click the "Expand" button to see more details about the moderation (see screenshot below). In this new page you can see a table with all the reports that belongs to that moderation.
- In the reports table there is an excerpt of the reported content in the language that the user was using at the moment. You can also click the "Expand" button to see the whole content. 

Both views includes a "preview" of the reported content. Since we don't want to display the whole content I created a new cell called `reported_content_cell` that can be configured for each `resource`. This cell will be used to preview this content in the email sent to the moderators.

#### Testing

1. Report some resources and if possible a few times (with different users).
1. Navigate to the admin dashboard and go to the moderations panel.
1. Click the "Expand" button to see the moderation details.

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.


### :camera: Screenshots

**Before:**

![image](https://user-images.githubusercontent.com/106021/96575931-b3dcf800-12d1-11eb-90e8-c880f07ba984.png)

**After:**

![image](https://user-images.githubusercontent.com/106021/96575982-c1927d80-12d1-11eb-8496-aa0d60fe7611.png)

![image](https://user-images.githubusercontent.com/106021/96575995-c7885e80-12d1-11eb-864e-cc1992fb3956.png)

_(previous screenshot when scrolling down a bit...)_

![image](https://user-images.githubusercontent.com/106021/96576312-49788780-12d2-11eb-90e4-656a0457fce0.png)

![image](https://user-images.githubusercontent.com/106021/96576352-5bf2c100-12d2-11eb-9543-4236228c820c.png)
